sycl/doc/extensions/experimental/sycl_ext_oneapi_bfloat16_math.asciidoc

```diff
@@ -0,0 +1,139 @@
+= sycl_ext_oneapi_bfloat16_math
```

Comment:
General comments on this file:

* It should be written to use the [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc).
* There should be a feature-test macro associated with this extension (`SYCL_EXT_ONEAPI_BFLOAT16_MATH`).
* Since these functions are not implemented on all devices, there should be some aspect to indicate when they can be called. I think we may as well move the `ext_oneapi_bfloat16` aspect to this extension, since that is what applications used to use to tell if these math functions can be called.  Ideally, the aspect would return `true` only on the CUDA devices where they are supported, but that doesn't need to be done in this PR.  We can instead change the aspect back to its original behavior where it always returned `false`.

Reply:
I thought I had used the template, but I was wrong, It uses the template now.
OK, on the feature test macro.
OK, we will use the old bfloat16 aspect to refer to the math functions.

Reply:
Thanks for updating this file to use the template.  It looks very good now.  Since this new extension uses the `aspect::ext_oneapi_bfloat16` aspect, we can no longer document this aspect as "deprecated" in the other extension.  It makes no sense for one extension to say this aspect is deprecated while another extension says it is non-deprecated.

Let's fix this by removing the section titled "Deprecated Features" from "sycl_ext_oneapi_bfloat16.asciidoc".

I also noticed that the example in "sycl_ext_oneapi_bfloat16.asciidoc" still uses the aspect.  Sorry for not catching this earlier.  The example currently looks like this:

```
  if (dev.has(aspect::ext_oneapi_bfloat16)) {
    deviceQueue.submit([&](handler &cgh) {
      accessor numbers{buf, cgh, read_write};
      cgh.single_task([=]() { numbers[2] = foo(numbers[0], numbers[1]); });
    });
  } else {
    std::cout << "No bfloat16 support\n";
    return 1;
  }
```

It should instead look like:

```
  deviceQueue.submit([&](handler &cgh) {
    accessor numbers{buf, cgh, read_write};
    cgh.single_task([=]() { numbers[2] = foo(numbers[0], numbers[1]); });
  });
```

Reply:
Yes, corrected now.